doorstop/core/item.py

```diff
@@ -188,7 +189,10 @@ def new(
 
         """
         UID(uid).check()
-        filename = str(uid) + Item.EXTENSIONS[0]
+        fileext = Item.EXTENSIONS[0]
+        if document.itemformat == "yamlfm":
+            fileext = ".md"
+        filename = str(uid) + fileext
```

Comment:
I propose renaming `itemformat` to `extension` (and defaulting to `."yml"`) so this code can be simplified to:

```
filename = str(uid) + document.extension
```

Reply:
The whole implementation has changed because the extensions where also used to identify valid files. Just adding to the `EXTENSIONS` array would lead to false positive file identifications (accept `.md` although format was `yaml`).